prov/sockets/src/sock_av.c

```diff
@@ -66,9 +66,9 @@ int sock_av_get_addr_index(struct sock_av *av, struct sockaddr_in *addr)
 	int i;
 	struct sock_av_addr *av_addr;
 
-	for (i = 0; i < av->table_hdr->stored; i++) {
+	for (i = 0; i < av->table_hdr->size; i++) {
 		av_addr = &av->table[i];
-		if (ofi_equals_sockaddr(addr, (struct sockaddr_in *)&av_addr->addr))
+		if (av_addr->valid && ofi_equals_sockaddr(addr, (struct sockaddr_in *)&av_addr->addr))
```

Comment:
Maybe this would be better expressed as separate statements:

```c
if (!av_addr->valid)
    continue;

if (ofi_equals_sockaddr(addr, (struct sockaddr_in *)&av_addr->addr))
    return i;

```